src/components/swap/components/transaction-review/provider.tsx

```diff
@@ -100,15 +100,17 @@ export function useTransactionReview(props: ReviewProps) {
 
   // const { simulateTrade } = useSimulateQuote(quote?.tx ?? null)
 
-  const refId = useRefId()
+  const { source, medium } = useUtmParams()
 
   const client = usePublicClient()
   const queryClient = useQueryClient()
   const saveTrade: TradeCallback = useCallback(
     async ({ address, hash, quote }) => {
       await fetch(`/api/user/trade`, {
         method: 'POST',
-        body: JSON.stringify(mapQuoteToTrade(address, hash, quote, refId)),
+        body: JSON.stringify(
+          mapQuoteToTrade(address, hash, quote, source, medium),
```

Comment:
so `refId` won't be stored any longer? or it's essentially the same now?

Reply:
It's the utm source, I just though since we'll have all utm tags now, ill call them by name, and refId will be mapped from the utm_source